llvm/lib/IR/Type.cpp

```diff
@@ -838,12 +838,14 @@ static TargetTypeInfo getTargetTypeInfo(const TargetExtType *Ty) {
     return TargetTypeInfo(PointerType::get(C, 0), TargetExtType::CanBeGlobal);
```

Comment:
I made a complete guess here that spirv.Image should not be used in allocas, so that I could use it as a test case. Please let me know if I got that wrong.

Reply:
Maybe @jcranmer-intel can comment on which `spirv.` types can/cannot be used in allocas.

Reply:
There are now some tests which do use spirv.Image in allocas:
```
test/CodeGen/SPIRV/transcoding/image_with_access_qualifiers.ll
test/CodeGen/SPIRV/transcoding/get_image_num_mip_levels.ll
test/CodeGen/SPIRV/image-unoptimized.ll
test/CodeGen/SPIRV/read_image.ll
```
@jcranmer-intel any comments on the validity of this? Is it a supported use of spirv.Image, or just something that was conventient for these test cases?

Reply:
Because of this I don't have any non-CanBeLocal types to test with, so the tests I added in `type/Assembler/target-type-properties.ll` are currently failing.

Reply:
I'm now using the newly added `amdgcn.named.barrier` for testing.